packages/orm/src/QueryParser.ts

```diff
@@ -86,6 +86,8 @@ export interface ParsedFindQuery {
   condition?: ParsedExpressionFilter;
   /** Any optional orders to add before the default 'order by id'. */
   orderBys: ParsedOrderBy[];
+  /** Optional CTE to prefix to the query, i.e. for recursive relations. */
+  cte?: { sql: string; bindings: readonly any[] };
```

Comment:
This is kind of "bag on the side", but adding a "blob of CTE sql" to `ParsedFindQuery`...

The idea is that we should be able to use the CTE basically as a fancy `where` clause, and then otherwise reuse the "how to conditionally join & loader CTI tables" that `ParsedFindQuery` can already represent.